plugins/parcelize/parcelize-compiler/parcelize.backend/src/org/jetbrains/kotlin/parcelize/IrParcelSerializerFactory.kt

```diff
@@ -155,6 +155,14 @@ class IrParcelSerializerFactory(private val symbols: AndroidSymbols) {
                 return ulongArraySerializer
         }
 
+        // Special handled library types
+        when (classifierFqName) {
+            "kotlin.time.Duration" ->
+                return wrapNullableSerializerIfNeeded(
+                    irType, IrDurationSerializer(classifier)
```

Comment:
Looking at the code now it occurs to me that we don't have to write our own serializer and deserializer for this. We already have a serializer that deals with value classes. It is the `IrUnsafeCoerceWrapperSerializer`. It uses the serializer for the underlying type and then just treats the result as an unboxed instance of the value class type. We should be able to use that directly here and treat `kotlin.duration.time` exactly as we treat `kotlin.ulong` which is also a value class with a long underlying type.

Reply:
Done